src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs

```diff
@@ -162,7 +200,15 @@ internal void AddHeaders(HttpRequestMessage httpRequest, string invocationId)
         {
             httpRequest.Headers.Add(HttpWorkerConstants.HostVersionHeaderName, ScriptHost.Version);
             httpRequest.Headers.Add(HttpWorkerConstants.InvocationIdHeaderName, invocationId);
-            httpRequest.Headers.UserAgent.ParseAdd($"{HttpWorkerConstants.UserAgentHeaderValue}/{ScriptHost.Version}");
+            httpRequest.Headers.UserAgent.ParseAdd(_userAgentString);
+        }
+
+        private void AddProxyingHeaders(HttpRequest httpRequest, string invocationId)
+        {
+            // if there are existing headers, override them
+            httpRequest.Headers[HttpWorkerConstants.HostVersionHeaderName] = ScriptHost.Version;
```

Comment:
Curious why we are not following the same '.Add' pattern as above? Is it because we're overriding? If not could we use Add or AndOrUpdate (if exists)?